lib/iris/fileformats/netcdf.py

```diff
@@ -1793,15 +1791,15 @@ def _get_cube_variable_name(self, cube):
 
     def _get_coord_variable_name(self, cube, coord):
         """
-        Returns a CF-netCDF variable name for the given coordinate.
+        Returns a CF-netCDF variable name for a given coordinate-like element.
 
         Args:
 
         * cube (:class:`iris.cube.Cube`):
             The cube that contains the given coordinate.
-        * coord (:class:`iris.coords.Coord`):
-            An instance of a coordinate for which a CF-netCDF variable
-            name is required.
+        * coord (:class:`iris.coords.DimensionalMetadata`):
```

Comment:
No, they really _are_ all ultimately _DimensionalMetadata, by inheritance
 \-- though I may have missed out a dunder ??

This is to be used with Coord, CellMeasure and Ancillary, 
and in future will add mesh coordinates (AuxCoord) and connectivities (Connectivity)
Inheritance is:
  * coords.Coord -> _DimensionalMetadata
  * coords.AncillaryVariable -> _DimensionalMetadata
  * coords.CellMeasure -> AncillaryVariable (-> _DimensionalMetadata)
  * ugrid.Connectivity -> _DimensionalMetadata

Reply:
Actually, a lot of the legacy code is still use 'coord' pretty loosely, where it now deals with other ~similar types of data. That is both in variable naming and docstrings.

If you think this is important, we can have a general blitz on it.  But I suggest we leave it until the last, so as to avoid generating a blizzard of extra changes to review.

Reply:
@pp-mo Cool. Understood. Thanks :+1: 

So it's `iris.coords._DimensionalMetadata` then, and not `iris.coords.DimensionalMetadata`, which doesn't exist.

Reply:
:+1: will check that out + fix it..